mut/redirects/redirect_main.py

```diff
@@ -0,0 +1,335 @@
+"""
+Usage:
+    mut-redirects <source_path> -o <output>
+
+    -h, --help             List CLI prototype, arguments, and options.
+    <source_path>          Path to the file(s) containing redirect rules.
+    -o, --output <output>  File path for the output .htaccess file.
+"""
+
+# Spec URL: https://docs.google.com/document/d/1oI2boFmtzvbbvt-uQawY9k_gLSLbW7LQO2RjVkvtRgg/edit?ts=57caf48b#
+
+import re
+import collections
+from typing import List
+from docopt import docopt
+
+RuleDefinition = collections.namedtuple('RuleDefinition', ('is_temp', 'version', 'old_url', 'new_url', 'is_symlink'))
+
+
+class RedirectContext:
+    def __init__(self, rules: List = [], symlinks: List = [], definitions: List = []):
+        self.rules = []
+        self.symlinks = []
+        self.definitions = []
+
+        return None
+
+    def add_definition(self, key: str, value: str):
+        d = [key, value]
+        self.definitions.append(d)
+
+    def generate_rule(self, is_temp: bool, version: str, old_url: str, new_url: str, is_symlink: bool = False):
+        context_url = ''
+
+        for x in range(0, len(self.definitions)):
+            if (self.definitions[x][0] == 'base'):
+                context_url = self.definitions[x][1]
+
+        # if url contains {version} - substitute in the correct version
+        old_url_sub = self.rule_substitute(old_url, version)
+        new_url_sub = self.rule_substitute(new_url, version)
+
+        # reformatting the old url
+        if len(old_url_sub) > 0:
+            if old_url_sub[0] != '/':
+                old_url_sub = '/' + old_url_sub
+
+            if not is_symlink:
+                if old_url_sub[-1] == '/':
+                    old_url_sub = old_url_sub[:-1]
+                old_url_sub = '/' + context_url + old_url_sub
+
+        new_rule = RuleDefinition._make([is_temp, version, old_url_sub, new_url_sub, False])
```

Comment:
No need to use `_make` here. Just `RuleDefinition(is_temp, version, old_url_sub, new_url_sub, False)`